awsapilib/sso/entities/entities.py

```diff
@@ -197,17 +210,34 @@ def instance_id(self):
             instance_id (str): The instance id of the account
 
         """
-        if self._instance_id is None:
-            account_id = self.id
-            target = 'com.amazon.switchboard.service.SWBService.GetApplicationInstanceForAWSAccount'
-            payload = self._sso.get_api_payload(content_string={'awsAccountId': account_id},
-                                                target='GetApplicationInstanceForAWSAccount',
-                                                path='/control/',
-                                                x_amz_target=target)
-            self.logger.debug('Trying to get instance id for aws account with payload: %s', payload)
-            response = self._sso.session.post(self.url, json=payload)
-            self._instance_id = response.json().get('applicationInstance', {}).get('instanceId', '')
-        return self._instance_id
+        instance_id = self._retrieve_instance_id()
+        if not instance_id:
+            instance_id = self._provision_application_instance_for_aws_account()
+        return instance_id
+
+    def _provision_application_instance_for_aws_account(self):
+        target = 'com.amazon.switchboard.service.SWBService.ProvisionApplicationInstanceForAWSAccount'
+        payload = self._sso.get_api_payload(content_string={'accountId': self.id,
+                                                            'accountEmail': self.email,
+                                                            'accountName': self.name
+                                                            },
+                                            target='ProvisionApplicationInstanceForAWSAccount',
+                                            path='/control/',
+                                            x_amz_target=target)
+        self.logger.debug('Trying to get instance id for aws account with payload: %s', payload)
+        response = self._sso.session.post(self.url, json=payload)
+        return response.json().get('applicationInstance', {}).get('instanceId', None)
+
+    def _retrieve_instance_id(self):
+        account_id = self.id
+        target = 'com.amazon.switchboard.service.SWBService.GetApplicationInstanceForAWSAccount'
+        payload = self._sso.get_api_payload(content_string={'awsAccountId': account_id},
+                                            target='GetApplicationInstanceForAWSAccount',
+                                            path='/control/',
+                                            x_amz_target=target)
+        self.logger.debug('Trying to get instance id for aws account with payload: %s', payload)
+        response = self._sso.session.post(self.url, json=payload)
+        return response.json().get('applicationInstance', {}).get('instanceId', None)
```

Comment:
Should we maybe here not assume that the response will always be succesful and check it before trying to access the json?